### ISSUE TITLE
Essentials Smart Home Premium Radiator Thermostat

### ISSUE THREAD

**Issue by cova-fe** [dresden-elektronik/deconz-rest-plugin#5386]
## Device

- Product name: Essentials Smart Home Premium Radiator Thermostat
- Manufacturer: _TZE200_jeaxp72v
- Model identifier: TS0601
- Device type : Please remove all unrelated device types. 
  - Thermostat

### Basic
![image](https://user-images.githubusercontent.com/385249/137633827-b4c4beb0-bbca-45ee-9cdc-8f35a01b8ed6.png)
![image](https://user-images.githubusercontent.com/385249/137634086-33f4566b-40da-4e02-8abe-2edb86d4b3ab.png)

### Groups
![image](https://user-images.githubusercontent.com/385249/137633938-65519706-8801-441f-b9c9-911ad5d4f343.png)

### Scenes
![image](https://user-images.githubusercontent.com/385249/137633964-9b222c76-f56f-4800-b6d2-98e8e4ab9fac.png)

**Comment by Mimiix** [comment]
Seems to be supported already? Doesn't it show In the rest api?

**Comment by cova-fe** [comment]
I did not noticed the case statement, right. I guess my change did not had adverse effect anyway. Regarding the phoscon web app, is there anything that could be done to have the device shown under sensors?

**Comment by cova-fe** [comment]
I have another thermostat, different brand, and it appears on the Phoscon app:
![image](https://user-images.githubusercontent.com/385249/137864885-8e6ef2fa-00e5-465a-b8cb-1ff4a3e48d4a.png)
so I was expecting something similar.

I will be happy to open a PR, unfortunately the device is not working as I expect (i.e. offset setting does not work yet). I'm looking around for more details on how this specific device, to see if I can improve the situation.

**Comment by Smanar** [comment]
Arf it's my fault.
I forget that.

On temperature cluster you have

```
                                    ResourceItem *item2 = i->item(RConfigOffset);
                                    if (item2 && item2->toNumber() != 0)
                                    {
                                        temp += item2->toNumber();
                                    }
```

But I forget it on tuya.cpp file.
I can explain how to do if you want to make a PR, but I m not sure wich one tuya command use your device, can be

```
case 0x0203: // Thermostat current temperature
case 0x0218: // Thermostat current temperature for moe
case 0x026B: // Temperature
```

**Comment by appleonkel** [comment]
@cova-fe 
The description says the thermostat has the windowopen_set at True in the initial state. If it detects an "open window" it will close the ventil for 30 minutes. But i am not able to detect this attribute in deconz/phoscon.
Another thing that is missing is the child-proof-lock.

I can see that sometime config/on is automatic set to true or false. And i am able to set the heatsetpoint.

**Comment by appleonkel** [comment]
This is what i get from deconz 2.12.06-raspbian-buster-stable
```
{
    "config": {
        "heatsetpoint": 2250,
        "offset": 0,
        "on": true,
        "reachable": true,
        "schedule": {},
        "schedule_on": false
    },
    "ep": 1,
    "etag": "577b8d9ac1b2d3d36979363cdc159207",
    "lastannounced": null,
    "lastseen": "2021-10-20T00:26Z",
    "manufacturername": "_TZE200_jeaxp72v",
    "modelid": "TS0601",
    "name": "Thermostat 23",
    "state": {
        "lastupdated": "2021-10-19T22:19:22.435",
        "on": false,
        "temperature": 2190
    },
    "type": "ZHAThermostat",
    "uniqueid": "84:71:27:ff:fe:73:04:b2-01-0201"
}
```
If someone can give me the right commands, i have the [cli-plugn](https://github.com/ma-ca/deconz-cli-plugin) running and could help debug with that.

**Comment by Smanar** [comment]
Ha ok, so it s normal, you need code modification to have full support.

**Comment by svn2208** [comment]
If I change the temperature, it won‘t affect the displayed temperature. Someone knows why?

**Comment by Mimiix** [comment]
@smanar

**Comment by Smanar** [comment]
Can you show your device json pls ?
Or debug logs when making the request ? With flags "info" and "info_l2" ?

**Comment by svn2208** [comment]
Do you mean this:

`“20": {
        "config": {
            "battery": 99,
            "heatsetpoint": 1600,
            "offset": 0,
            "on": true,
            "reachable": true,
            "schedule": {},
            "schedule_on": false
        },
        "ep": 1,
        "etag": "1e8b9781d9584dc7ec9fa011390f1a41",
        "lastannounced": "2021-12-06T19:49:02Z",
        "lastseen": "2021-12-09T17:34Z",
        "manufacturername": "_TZE200_jeaxp72v",
        "modelid": "TS0601",
        "name": "Thermostat 20",
        "state": {
            "lastupdated": "2021-12-09T17:31:02.117",
            "on": false,
            "temperature": 1880
        },
        "type": "ZHAThermostat",
        "uniqueid": "84:fd:27:ff:fe:16:50:5c-01-0201"
    }
}`


Info and info l2 are: —dpg-info=2?

**Comment by svn2208** [comment]
#5404 / #5405 solved my problem.

**Comment by Smanar** [comment]
Yep, the device was missing in code, new manufacture name > _TZE200_jeaxp72v

**Comment by Mimiix** [comment]
@Smanar is battery integrated for this device?

**Comment by appleonkel** [comment]
@Mimiix Yes battery is integrated and as far as i can say, the values from the API are correct.

**Comment by svn2208** [comment]
It is not working for my device.
![CD9D5C28-0DC3-4BF1-BE7E-FCCECCCADA5A](https://user-images.githubusercontent.com/8638258/148290291-0d889036-065e-47cf-8fc9-49e0b7f13e8b.jpeg)

**Comment by svn2208** [comment]
I could resolve this by reading the node descriptor. Thank anyway

**Comment by scarre** [comment]
Is the fix available in deconz 2.13.4 ?
I have this thermostat with manufacturer _TZE200_jeaxp72v, but it works only in one direction: temperature changes at the thermostat itself seems to be handled in deconz. Temperature changes in deconz are not taken into account by the thermostat itself. Other thermostat features do not work.
If not in 2.13.4, will the fix be available in 2.13.5 ?
![TZ200_jeaxp72v_nodeinfo](https://user-images.githubusercontent.com/2837872/150183723-4a242a79-d6ee-45e1-9873-454a62b35ae5.png)

**Comment by wallesch** [comment]
Running HA on supervised installation with deCONZ, cannot using BETA ... waiting over months to get _TZE200_jeaxp72v available ... Release date moved from November => December  ... all my hope on 2.13.4 and AddOn 6.11 ...  but nothing changed ... sometimes i think HA and deCONZ  was a wrong decision

**Comment by svn2208** [comment]
I compiled it by myself. Maybe it will work for you as well. Just exchange the lib then

**Comment by Smanar** [comment]
I don't see the model "_TZE200_jeaxp72v" on the code.
And yes ATM all PR are locked to don't make the code too time, as in the future we will use DDF instead of c++ edition.

**Comment by scarre** [comment]
This is strange, because this device is a clone of  {"_TYST11_jeaxp72v", "eaxp72v", "Essentials", "Tuya_THD Essentials TRV"} (same manufacturer wesmartify/essentials, same model, just a different reference) which thus only requires an additional 1 line of code.
Such cloned devices were committed about 2 months ago (#5497), while this one wasn't (#5404) even though it has been identified as not conflicting since october 21 ?
I guess I'll try to compile myself and replace the lib, as suggested by [svn2208](https://github.com/svn2208)...

**Comment by scarre** [comment]
@svn2208 
> I compiled it by myself. Maybe it will work for you as well. Just exchange the lib then
I successfully modified the sources of v2.13.4, recompiled and replaced the lib. Thermostat is recognized properly now.
The only thing that does not work for me is the "heat" indication it always displays "off" even when thermostat is in heating mode. Otherwise all other functionality is working well.
Do you have the same experience with the "heat" indication always indicated off ?

**Comment by svn2208** [comment]
Hmm I have the feeling it is alway on heat

**Comment by Smanar** [comment]
You have a config/mode on your device ? it seem from the code there is only a config/preset  (at least for the _TYST11_jeaxp72v)

**Comment by scarre** [comment]
Ok, can you guide me a bit on where to find this info for config/mode ?
Before I recompiled deConz v2.13.4, the node had a thermostat cluster entry 0201, but the thermostat cluster info was empty after executing a read (all lines grayed out).
With the recompiled version, the node no longer has the thermostat cluster entry (see image), but behaves correctly as a thermostat, apart from the heat on/off status.
Here is the json from deConz api:
{"config":{"battery":97,"heatsetpoint":1500,"locked":false,"offset":0,"on":true,"preset":"manual","reachable":true,"schedule":{},"schedule_on":false,"setvalve":true,"windowopen_set":true},"ep":1,"etag":"2f98774485ee76f73efd71bdab487a33","lastannounced":"2022-01-21T08:14:26Z","lastseen":"2022-01-24T10:13Z","manufacturername":"_TZE200_jeaxp72v","modelid":"TS0601","name":"Thermostat 27","state":{"lastupdated":"2022-01-24T09:44:26.837","on":false,"temperature":2160},"type":"ZHAThermostat","uniqueid":"84:fd:27:ff:fe:a2:4c:17-01-0201"}
I use Home Assistant on a second raspberry pi to control my Phoscon Gateway. In HA, the heat indicator is always off, even when the radiator is physically heating, so it probably reads this info from the "on" field in the reported state ? 
![TZ200_jeaxp72v_node_deconz_modified](https://user-images.githubusercontent.com/2837872/150764837-fa2e8824-7229-42c9-8a62-9d932fd1524e.png)

**Comment by Smanar** [comment]
```
{
  "config": {
    "battery": 97,
    "heatsetpoint": 1500,
    "locked": false,
    "offset": 0,
    "on": true,
    "preset": "manual",
    "reachable": true,
    "schedule": {},
    "schedule_on": false,
    "setvalve": true,
    "windowopen_set": true
  },
  "ep": 1,
  "etag": "2f98774485ee76f73efd71bdab487a33",
  "lastannounced": "2022-01-21T08:14:26Z",
  "lastseen": "2022-01-24T10:13Z",
  "manufacturername": "_TZE200_jeaxp72v",
  "modelid": "TS0601",
  "name": "Thermostat 27",
  "state": {
    "lastupdated": "2022-01-24T09:44:26.837",
    "on": false,
    "temperature": 2160
  },
  "type": "ZHAThermostat",
  "uniqueid": "84:fd:27:ff:fe:a2:4c:17-01-0201"
}
```

So yes I have some explanations:
- your device don't use config/mode, so HA use config/on and this field is always true
- You have config/preset that can be one of thoses values
```
const std::array<KeyValMapTuyaSingle, 7> RConfigPresetValuesTuya = { { {QLatin1String("holiday"), {0x00}}, {QLatin1String("auto"), {0x01}}, {QLatin1String("manual"), {0x02}},
                                                                              {QLatin1String("comfort"), {0x04}}, {QLatin1String("eco"), {0x05}}, {QLatin1String("boost"), {0x06}},
                                                                              {QLatin1String("complex"), {0x07}} } };
```
If I remember the "off" is preset = holidays

AND BTW your device is not yet in the code, need this PR https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5404

**Comment by Smanar** [comment]
>Any idea why the device thermostat cluster (0201) for this device does not appear in deConz GUI after patching the code as proposed in #5404 (see image in my post above) ?

Not realy, tuya sometime hide the cluster, so need deconz detect it to display it, but in all case, this device don't use it and use instead the tuya cluster, so I can't explain the result, but you lost nothing.

**Comment by scarre** [comment]
@bmgit26 (from #5745)
#I compiled the not yet merged branch from https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5404.
I see more infos now from the Rest-API coming from deconz/thermostat, but I still can't set a new heatpoint that is accepted by the thermostat:

Make sure that you physically have the thermostat on manual (displays a red hand below the temperature), then it will respond to changes of heatset point from deconz. If physically in schedule or anti-freeze mode, it does not respond.

**Comment by bmgit26** [comment]
Yes, thanks, this is working. Even setting the mode via restapi/home
assistant, and then setting the Heatpoint.

pingu077 ***@***.***> schrieb am Sa., 5. Feb. 2022, 12:12:

**Comment by Floyer007** [comment]
Works for me now as well with https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5404
Manual-mode did the trick.

How did you disabled the window-open detection @appleonkel?

**Comment by scarre** [comment]
> How did you disabled the window-open detection
curl -X PUT -d "{\"windowopen_set\": false}" http://your_deconz_ip/api/your_api_key/sensors/your_sensor_id/config

**Comment by herrmarten** [comment]
hey @scarre i am missing exactly that confi "windowopen_set" but my thermostat shows the window icon. do you have an idea how i can get it under control?

**Comment by scarre** [comment]
You are also missing "locked", "offset", "preset", "setvalve".
Have you recompiled and replaced the deconz lib with the changes from #5404 ?

**Comment by herrmarten** [comment]
thanks for your response @scarre  sorry, i am super new to this. i am running deconz in a docker-container. guess i will switch to install it directly or is there a way to get #5405 on docker?

**Comment by bmgit26** [comment]
BTW, is the current configuration from #5404 meant to stay like that?

I mean everything is "working", but in HomeAssistant there are quite a few glitches that make it differ from other thermostats:
- you need to manually switch mode in settings to manual
- even when this is done, the status is never correctly shown in homeassistant

Is this on HomeAssistant or deconz? I tend to say deconz, as my other thermostats work nicely in homeassistant.

Thanks
Max

**Comment by herrmarten** [comment]
@scarre thank you so much again for your advises. i tried around, compiling the restplugin etc. but i ran in multiple different problems even after compiling successfully. i ended up unpairing all devices by accident by starting zigbee2mqtt to change the windowopen_set via zigbee2mqtt (which worked) ... i just finished pairing the devices i had paired to deconz before with zigbee2mqtt. luckily it were just a few. most is still on huebridge (as i said, i am quite new to using an own solution)

**Comment by bmgit26** [comment]
Any idea why this thermostat is not behaving as others in Home Assistant? (https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5386#issuecomment-1034654864)

**Comment by Smanar** [comment]
Hello.
The DDF core will be able soon to support Tuya cluster, so I don't think the PR will be merged.

But If you want to try using DDF I can help you, but you will be the first one, not sure how far we can go.
You need a recent deconz version, the new code is not yet in stable https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5868

**Comment by Smanar** [comment]
Yep it was in it.

So can start a DDF, as it will be nor reconized at all, you will have to set all the DDF.
You have a video here https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/DDF-cheat-sheet

- Just make a DDF with filling the "basic cluster identification"
- drag and drop a ZHAThermostat device
- set the status to gold
- save the file
-  Make hot reload

Now the device will be reconised as thermostat, but not working at all.

Can you copy paste the DDF, I will edit it to add the temperature and the offset (I will try at least)

**Comment by bmgit26** [comment]
Hey @Smanar, thanks a lot, I will definitely give this a try but dont know exactly when i can get to this. Most likely on Friday or the weekend.

**Comment by Smanar** [comment]
There is a PR in waiting list to support tuya TRV using DDF.

Getting another device is hard because tuya make so much clone, almost impossible to be sure having the good one.
On the code as supported natively Tuya TRV there is

```
    // Tuya Thermostat / TRV
    {"_TYST11_zuhszj9s", "uhszj9s", "HiHome", "Tuya_THD WZB-TRVL TRV"},
    {"_TZE200_zuhszj9s", "TS0601", "HiHome", "Tuya_THD WZB-TRVL TRV"},
    {"_TYST11_KGbxAXL2", "GbxAXL2", "Saswell", "Tuya_THD SEA801-ZIGBEE TRV"},
    {"_TYST11_c88teujp", "88teujp", "Saswell", "Tuya_THD SEA801-ZIGBEE TRV"},
    {"_TZE200_c88teujp", "TS0601", "Saswell", "Tuya_THD SEA801-ZIGBEE TRV"},
    {"_TYST11_ckud7u2l", "kud7u2l", "Tuya", "Tuya_THD HY369 TRV"},
    {"_TZE200_ckud7u2l", "TS0601", "Tuya", "Tuya_THD HY369 TRV"},
    {"_TZE200_ywdxldoj", "TS0601", "MOES/tuya", "Tuya_THD HY368 TRV"},
    {"_TZE200_fhn3negr", "TS0601", "MOES/tuya", "Tuya_THD MOES TRV"},
    {"_TZE200_aoclfnxz", "TS0601", "Moes", "Tuya_THD BTH-002 Thermostat"},
    {"_TYST11_jeaxp72v", "eaxp72v", "Essentials", "Tuya_THD Essentials TRV"},
    {"_TZE200_jeaxp72v", "TS0601", "Essentials", "Tuya_THD Essentials TRV"},
    {"_TYST11_kfvq6avy", "fvq6avy", "Revolt", "Tuya_THD NX-4911-675 TRV"},
    {"_TZE200_kfvq6avy", "TS0601", "Revolt", "Tuya_THD NX-4911-675 TRV"},
    {"_TYST11_zivfvd7h", "ivfvd7h", "Siterwell", "Tuya_THD GS361A-H04 TRV"},
    {"_TZE200_zivfvd7h", "TS0601", "Siterwell", "Tuya_THD GS361A-H04 TRV"},
    {"_TYST11_yw7cahqs", "w7cahqs", "Hama", "Tuya_THD Smart radiator TRV"},
    {"_TZE200_yw7cahqs", "TS0601", "Hama", "Tuya_THD Smart radiator TRV"},
    {"_TZE200_cwnjrr72", "TS0601", "MOES", "Tuya_THD HY368 TRV"},
    {"_TZE200_b6wax7g0", "TS0601", "MOES", "Tuya_THD BRT-100"},
```


Else can use TRV but from other brand than tuya

**Comment by WhiteBahamut** [comment]
tl;dr:
Not sure if people still look for that. My ConBee II detect the thermostat, but does *not* show it in the sensors (or anywhere else).

I can access the sensor via the API and set config values. Also HA detects the sensor and I can set temperatures. In HA the "heat" and "on/off" button do not seem to work. Also switching to preset modes seems doable. Tho I dont know what the buttons and the modes should ;D
It would be great if the Phoscon PWA would also show the sensor. I had 3 tries until I realized it was added but just not displayed  (added via the PWA).
For completeness I post my sensor
```json
{
  "config": {
    "battery": 100,
    "heatsetpoint": 700,
    "locked": false,
    "offset": 0,
    "on": true,
    "preset": "auto",
    "reachable": true,
    "schedule": {},
    "schedule_on": false,
    "setvalve": true,
    "windowopen_set": true
  },
  "ep": 1,
  "etag": "xxxxxx",
  "lastannounced": "2022-09-28T20:46:26Z",
  "lastseen": "2022-09-29T13:55Z",
  "manufacturername": "_TZE200_jeaxp72v",
  "modelid": "TS0601",
  "name": "Thermostat 3",
  "state": {
    "lastupdated": "2022-09-29T12:46:54.682",
    "on": false,
    "temperature": 1760
  },
  "type": "ZHAThermostat",
  "uniqueid": "xxxxxx"
}
```

**Comment by Smanar** [comment]
```
  "manufacturername": "_TZE200_jeaxp72v",
  "modelid": "TS0601",
```

This device haven't the config/mode so action like heat/on/off are not available, but preset yes.
